src/components/OpenGovTreasuryProposal/CreatePreimage.tsx

```diff
@@ -0,0 +1,714 @@
+
+	};
+
+	const handleSubmit = async() => {
+
+		if(!isPreimage){if(txFee.gte(availableBalance)) return;}
+		await form.validateFields();
+		isPreimage && onChangeLocalStorageSet({ preimageLinked: true }, Boolean(isPreimage), preimageCreated, true);
+
+		if(!(isPreimage) ? preimageCreated : preimageLinked) {
+			setSteps({ percent: 100, step: 2 });
+		}
+		else{
+			!isPreimage ? await getPreimage() : (preimageLength !== 0 && beneficiaryAddress?.length > 0 && fundingAmount.gt(ZERO_BN)) && setSteps({ percent: 100, step: 2 }) ;
+			setEnactment({ ...enactment, value: enactment.key === EEnactment.At_Block_No ? advancedDetails?.atBlockNo : advancedDetails?.afterNoOfBlocks  });
+		}
+	};
+
+	const getExistPreimageDataFromPolkadot = async(preimageHash: string, isPreimage: boolean) => {
+		if(!api || !apiReady) return;
+
+		const lengthObj = await api.query.preimage.statusFor(preimageHash);
+
+		const length = JSON.parse(JSON.stringify(lengthObj))?.unrequested?.len || 0;
+		checkPreimageHash(length, preimageHash);
+		setPreimageLength(length);
+		form.setFieldValue('preimage_length', length);
+		onChangeLocalStorageSet({ preimageLength: length || '' }, Boolean(isPreimage));
+
+		const preimageRaw: any = await api.query.preimage.preimageFor([preimageHash, length ]);
+		const preimage = preimageRaw.unwrapOr(null);
+
+		const constructProposal = function(
+			api: ApiPromise,
+			bytes: Bytes
+		): Proposal | undefined {
+			let proposal: Proposal | undefined;
+
+			try {
+				proposal = api.registry.createType('Proposal', bytes.toU8a(true));
+			} catch (error) {
+				console.log(error);
+			}
+
+			return proposal;
+		};
+
+		try{
+			const proposal = constructProposal(api, preimage);
+			// eslint-disable-next-line @typescript-eslint/no-unused-vars
+			const { meta, method, section } = api.registry.findMetaCall(
+				proposal?.callIndex as any
+			);
+			const params = proposal?.meta ? proposal?.meta.args
+				.filter(({ type }): boolean => type.toString() !== 'Origin')
+				.map(({ name }) => name.toString()) : [];
+
+			const values = proposal?.args;
+
+			const preImageArguments = proposal?.args && params && params.map((name, index) => {
+				return {
+					name,
+					value: values?.[index]?.toString()
+				};
+			});
+			if(preImageArguments && proposal.section === 'Treasury' && proposal?.method === 'spend'){
+				const balance = new BN(preImageArguments[0].value || '0') || ZERO_BN;
+				setBeneficiaryAddress(preImageArguments[1].value || '');
+				setFundingAmount(balance);
+				onChangeLocalStorageSet({ beneficiaryAddress: preImageArguments[1].value || '', fundingAmount: balance.toString() }, Boolean(isPreimage));
+				setSteps({ percent: 100 ,step: 1 });
+				for(const i in maxSpendArr){
+					const [maxSpend] = inputToBn(String(maxSpendArr[i].maxSpend), network, false);
+					if(maxSpend.gte(balance)){
+						setSelectedTrack(maxSpendArr[i].track);
```

Comment:
do not set state on every iteration of the loop, store it in a var and set after the loop is completed